activestorage/app/controllers/active_storage/base_controller.rb

```diff
@@ -9,6 +9,17 @@ class ActiveStorage::BaseController < ActionController::Base
   self.etag_with_template_digest = false
 
   private
+    def stream_chunk(blob)
+      range = Rack::Utils.get_byte_ranges(request.headers["HTTP_RANGE"], blob.byte_size).first
+      chunk = blob.download_chunk range
+
+      response.header["Content-Range"] = "bytes #{range.begin}-#{range.end}/#{blob.byte_size}"
+      response.status = 206
```

Comment:
shouldn't we consider the error case responding `416` status code? https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests#partial_request_responses

Reply:
I initially thought this would be pretty hard to support since we'd really be dealing with how each underlying service handles the response.

Looking at it further, it looks like `Rack::Utils.get_byte_ranges` would return something like `[nil]` if there's an invalid range. I could add a check that and short circuit the it there. Something like:

```
def stream_chunk(blob)
  range = Rack::Utils.get_byte_ranges(request.headers["HTTP_RANGE"], blob.byte_size).first

  if range
    chunk = blob.download_chunk range

    response.header["Content-Range"] = "bytes #{range.begin}-#{range.end}/#{blob.byte_size}"
    response.status = 206
    response.stream.write chunk
  else
    response.status = 416
  end
ensure
  response.stream.close
end
```

Reply:
Nice! I think that makes sense